src/hotspot/os/posix/perfMemory_posix.cpp

```diff
@@ -92,12 +93,12 @@ static void delete_standard_memory(char* addr, size_t size) {
 //
 static void save_memory_to_file(char* addr, size_t size) {
 
- const char* destfile = PerfMemory::get_perfdata_file_path();
- assert(destfile[0] != '\0', "invalid PerfData file path");
+  const char* destfile = PerfMemory::get_perfdata_file_path();
+  assert(destfile[0] != '\0', "invalid PerfData file path");
 
   int result;
 
-  RESTARTABLE(os::open(destfile, O_CREAT|O_WRONLY|O_TRUNC, S_IRUSR|S_IWUSR),
+  RESTARTABLE(os::open(destfile, O_CREAT|O_WRONLY|O_TRUNC, S_IREAD|S_IWRITE),
```

Comment:
Using S_IREAD/S_IWRITE seems a bit odd when libc docs describe them as "obsolete synonym provided for BSD compatibility"!

Reply:
Changed to use S_IRUSR|S_IWUSR.